File: app/api/v1/endpoints/themes.py
from typing import List
from fastapi import Depends, APIRouter
from app.queries.themes import get_themes_by_subject
from app.schemas.theme import Theme as ThemeSchema
from app.config.database import get_db
from sqlalchemy.orm import Session
from app.utils.users import get_current_user

router = APIRouter(
    prefix="/subjects",
    tags=["subjects"],
)


@router.get("/{subject_id}/themes", response_model=List[ThemeSchema])
async def get_themes(
    subject_id: int, db: Session = Depends(get_db), user_id=Depends(get_current_user)
):
    return get_themes_by_subject(subject_id, user_id, session=db)


@router.post("/{subject_id}/themes", response_model=ThemeSchema)
async def create_theme(
    subject_id: int,
    theme: ThemeSchema,
    db: Session = Depends(get_db),
    user_id=Depends(get_current_user),
):

    return


@router.get("/{subject_id}/themes/{theme_id}", response_model=ThemeSchema)
async def get_theme(
    subject_id: int,
    theme_id: int,
    db: Session = Depends(get_db),
    user_id=Depends(get_current_user),
):

    return


@router.put("/{subject_id}/themes/{theme_id}", response_model=ThemeSchema)
async def update_theme(
    subject_id: int,
    theme_id: int,
    theme: ThemeSchema,
    db: Session = Depends(get_db),
    user_id=Depends(get_current_user),
):

    return


@router.delete("/{subject_id}/themes/{theme_id}")
async def delete_theme(
    subject_id: int,
    theme_id: int,
    db: Session = Depends(get_db),
    user_id=Depends(get_current_user),
):

    return


@router.patch("/{subject_id}/themes/{theme_id}", response_model=ThemeSchema)
async def patch_theme(
    subject_id: int,
    theme_id: int,
    theme: ThemeSchema,
    db: Session = Depends(get_db),
    user_id=Depends(get_current_user),
):

    return

File: frontend/src/api/themes.js
import apiClient from './client';

export const themesAPI = {
  // Получить все темы предмета
  list: async (subjectId) => {
    const response = await apiClient.get(`/api/v1/subjects/${subjectId}/themes`);
    return response.data;
  },

  // Получить одну тему
  get: async (subjectId, themeId) => {
    const response = await apiClient.get(`/api/v1/subjects/${subjectId}/themes/${themeId}`);
    return response.data;
  },

  // Создать тему
  create: async (subjectId, data) => {
    const response = await apiClient.post(`/api/v1/subjects/${subjectId}/themes`, data);
    return response.data;
  },

  // Обновить тему
  update: async (subjectId, themeId, data) => {
    const response = await apiClient.put(`/api/v1/subjects/${subjectId}/themes/${themeId}`, data);
    return response.data;
  },

  // Удалить тему
  delete: async (subjectId, themeId) => {
    await apiClient.delete(`/api/v1/subjects/${subjectId}/themes/${themeId}`);
  },
};